codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/EnumGenerator.kt

```diff
@@ -204,35 +205,44 @@ open class EnumGenerator(
                 }
             }
 
-            rust("/// Returns all the `&str` values of the enum members.")
-            rustBlock("pub const fn $Values() -> &'static [&'static str]") {
-                withBlock("&[", "]") {
-                    val memberList = sortedMembers.joinToString(", ") { it.value.doubleQuote() }
-                    write(memberList)
-                }
-            }
+            rust("#{T}", valuesMethod())
         }
     }
 
+    private fun valuesMethod() = writable {
+        rust(
+            """
+            /// Returns all the `&str` representations of the enum members.
+            pub const fn $Values() -> &'static [&'static str] {
+                &[$memberList]
+            }
+            """,
+        )
+    }
+
     private fun unknownVariantValue(): RuntimeType {
         return RuntimeType.forInlineFun(UnknownVariantValue, RustModule.Types) {
             docs(
                 """
                 Opaque struct used as inner data for the `Unknown` variant defined in enums in
-                the crate
+                the crate.
 
                 While this is not intended to be used directly, it is marked as `pub` because it is
                 part of the enums that are public interface.
-                """.trimIndent(),
+                """,
             )
             meta.render(this)
-            rust("struct $UnknownVariantValue(pub(crate) String);")
-            rustBlock("impl $UnknownVariantValue") {
-                // The generated as_str is not pub as we need to prevent users from calling it on this opaque struct.
-                rustBlock("pub(crate) fn as_str(&self) -> &str") {
-                    rust("&self.0")
+            // The generated `as_str` is not `pub` as we need to prevent users from calling it on this opaque struct.
+            rust(
+                """
+                struct $UnknownVariantValue(pub(crate) String);
+                impl $UnknownVariantValue {
+                    pub(crate) fn as_str(&self) -> &str {
+                        &self.0
+                    }
                 }
-            }
+                """,
+            )
         }
     }
```

Comment:
Can this be written as this (the syntax might be slightly off)?
```
rustTemplate("""
                /// Opaque struct used as inner data for the `Unknown` variant defined in enums in
                /// the crate.
                /// While this is not intended to be used directly, it is marked as `pub` because it is
                /// part of the enums that are public interface.
                #{Metadata:W}
                struct $UnknownVariantValue(pub(crate) String);
                impl $UnknownVariantValue {
                    pub(crate) fn as_str(&self) -> &str {
                        &self.0
                    }
                }
""",
"Metadata" to meta::render
)
```